src/sema/expression.rs

```diff
@@ -6965,6 +6965,77 @@ fn array_literal(
     let mut dims = Box::new(Vec::new());
     let mut flattened = Vec::new();
 
+    let resolve_to = match resolve_to {
+        ResolveTo::Type(Type::Array(elem_ty, _)) => ResolveTo::Type(elem_ty),
+        // Solana seeds are a slice of slice of bytes, e.g. [ [ "fo", "o" ], [ "b", "a", "r"]]. In this
```

Comment:
You could see it as a `bytes[][2]`, that's true.  We explicitly resolve the seeds as a slice of slice of bytes.

Reply:
We need slices because of the type required for the call, right? I was just wondering if, instead of slices, we could process the seeds as `bytes[][2]` to avoid adding new types to Solidity.

Reply:
Exactly, we need new types for the system call. 

Also, at some point we want to add support for `foo[5:6]` to Solidity too, which is a missing language feature right now. For that we would need to support slices.